src/javascript/JContent/ContentRoute/ToolBar/SortSelector/SortSelector.jsx

```diff
@@ -0,0 +1,113 @@
+import React, {useMemo} from 'react';
+import PropTypes from 'prop-types';
+import {shallowEqual, useDispatch, useSelector} from 'react-redux';
+import {useTranslation} from 'react-i18next';
+import {cmSetSort} from '~/JContent/redux/sort.redux';
+import {ArrowDown, ArrowUp, Dropdown, CustomDropdown, Separator, MenuItem} from '@jahia/moonstone';
+import classes from './SortSelector.scss';
+import JContentConstants from '~/JContent/JContent.constants';
+
+const SORT_DATA = [
+    {
+        label: 'jcontent:label.contentManager.listColumns.name',
+        orderBy: 'displayName'
+    },
+    {
+        label: 'jcontent:label.contentManager.listColumns.type',
+        orderBy: 'content.mimeType.value'
+    },
+    {
+        label: 'jcontent:label.contentManager.listColumns.createdBy',
+        orderBy: 'createdBy.value'
+    },
+    {
+        label: 'jcontent:label.contentManager.listColumns.lastModified',
+        orderBy: 'lastModified.value'
+    },
+    {
+        label: 'jcontent:label.contentManager.listColumns.size',
+        orderBy: 'content.data.size'
+    }
+];
+
+const DROPDOWN_ICON = {
+    DESC: <ArrowDown/>,
+    ASC: <ArrowUp/>
+};
+
+export const SortSelector = ({selector, setSortAction}) => {
+    const {t} = useTranslation('jcontent');
+
+    const {sort, mode} = useSelector(selector, shallowEqual);
+    const dispatch = useDispatch();
+    const sortData = useMemo(() => {
+        return SORT_DATA.map(d => ({label: t(d.label), value: d.orderBy}));
+    }, [t]);
+    const currentLabel = t(SORT_DATA.find(d => d.orderBy === sort?.orderBy)?.label);
+
+    return mode === JContentConstants.mode.GRID ? (
+        <CustomDropdown size="default"
+                        data-sel-role="sel-media-sort-dropdown"
+                        label={currentLabel}
+                        icon={DROPDOWN_ICON[sort?.order]}
+                        variant="outlined"
+        >
+            <MenuItem
+            className={classes.label}
+            label={t('jcontent:label.contentManager.sortSelector.orderingBy')}
+            variant="title"
+        />
+            <Dropdown
+                data-sel-role="sel-media-sort-property-dropdown"
+                data={sortData}
+                value={sort?.orderBy}
+                variant="outlined"
+                onChange={(e, item) => {
+                dispatch(setSortAction({order: sort.order, orderBy: item.value}));
+            }}
+        />
+            <Separator/>
+            <MenuItem
+            className={classes.label}
+            label={t('jcontent:label.contentManager.sortSelector.direction')}
+            variant="title"
+        />
+            <Dropdown
+                data-sel-role="sel-media-sort-order-dropdown"
+                variant="outlined"
+                value={sort?.order}
+                data={[
+                {
```

Comment:
we should also insert selector roles for each of the dropdown items using `attributes` data object (see `ContentStatusSelector` for example) and use it for the test selectors instead of using labels.